packages/react-dev-utils/ModuleScopePlugin.js

```diff
@@ -37,24 +37,21 @@ class ModuleScopePlugin {
       // Maybe an indexOf === 0 would be better?
       const relative = path.relative(appSrc, request.context.issuer);
       // If it's not in src/ or a subdirectory, not our request!
-      if (
-        relative.startsWith('../') ||
-        relative.startsWith('..\\')
-      ) {
+      if (relative.startsWith('../') || relative.startsWith('..\\')) {
         return callback();
       }
-      // Find path from src to the requested file
-      const requestRelative = path.relative(
-        appSrc,
-        path.resolve(
-          path.dirname(request.context.issuer),
-          request.__innerRequest_request
-        )
+      const requestFullPath = path.resolve(
+        path.dirname(request.context.issuer),
+        request.__innerRequest_request
       );
+      const requestRelative = path.relative(appSrc, requestFullPath);
+      if (/^(..[/|\\])+package(.json)?$/.test(requestRelative)) {
```

Comment:
@Timer I changed it to regex, I added '?' in  `(.json)?`  to make it optional. that way you can `require("../package")`